### DISCUSSION TITLE
Selecting a half rating with q-rating

### DISCUSSION THREAD

**Discussion by kamcnally** [quasarframework/quasar#12040] [General - Components / Directives / etc]
Is it possible to select half ratings when selecting with the mouse on the q-rating component?

The [floating number example](https://quasar.dev/vue-components/rating#floating-number) shows half ratings visually but only allows integer selection.

**Comment by pdanpdan** [comment]
Divide your model by 2

**Comment by DrewWalsh** [comment]
This will set it to a half rating when you click on the left half of the icon like how it works in Vuetify. Set the number in the if statement to half of the width of the rating icon SVG.

```
<q-rating
@click="rateItem($event)"
v-model="rating"
/>

...
rateItem(event) {
	const clickTarget = event.target;
	const xCoordInClickTarget = event.clientX - clickTarget.getBoundingClientRect().left;
	if (xCoordInClickTarget <= 30.665) {
		console.log("left");
		rating = rating - 0.5;
	} else {
		console.log("right");
	}
}

```
